spec/index.adoc

```diff
@@ -735,6 +740,13 @@ The error message is guaranteed to be a string, i.e. not arbitrary binary data.
 
 When canisters explicitly reject a message (see <<system-api-requests>>), they can specify the reject message, but _not_ the reject code; it is always `CANISTER_REJECT`. In this sense, the reject code is trustworthy: If the IC responds with a `SYS_FATAL` reject, then it really was the IC issuing this reject.
 
+[#error-codes]
+=== Error codes
+
+Implementations of the API can provide additional details for rejected messages in the form of a textual label identifying the error condition.
+API clients can use these labels to handle errors programmatically or suggest recovery paths to the user.
+The specification reserves error codes matching the regular expression `IC[0-9]+` (e.g., `IC502`) for the DFINITY implementation of the API.
```

Comment:
Do we plan to document the exact error codes somewhere?

Reply:
Yes, we will have short articles about each error code, and DFX will link to these articles. We aren't going to describe the errors in the spec for now.